qiita_pet/handlers/api_proxy/prep_template.py

```diff
@@ -6,52 +6,383 @@
+        the category and the number of times its seen.
+        Format {'status': status,
+                'message': message,
+                'num_samples': value,
+                'category': [(val1, count1), (val2, count2), ...], ...}
+    """
+    prep = PrepTemplate(int(prep_id))
+    access_error = check_access(prep.study_id, user_id)
+    if access_error:
+        return access_error
+    df = prep.to_dataframe()
+    out = {'num_samples': df.shape[0],
+           'summary': {}}
+
+    # drop the prep_id column if it exists
+    if 'study_id' in df.columns:
+        df.drop('study_id', axis=1, inplace=True)
+    cols = list(df.columns)
+    for column in cols:
+        counts = df[column].value_counts()
+        out['summary'][str(column)] = [(str(key), counts[key])
+                                       for key in natsorted(counts.index)]
+    # Add expected messaging and status info
+    out.update({'status': 'success', 'message': ''})
+    return out
 
 
-def prep_template_get_req(study_id, user_id):
-    """Gives a summary of each prep template attached to the study
+@execute_as_transaction
+def prep_template_post_req(study_id, user_id, prep_template, data_type,
+                           investigation_type=None,
+                           user_defined_investigation_type=None,
+                           new_investigation_type=None):
+    """Adds a prep template to the system
 
     Parameters
     ----------
     study_id : int
-        Study id to get prep template info for
+        Study to attach the prep template to
+    user_id : str
+        User adding the prep template
+    prep_template : str
+        Filepath to the prep template being added
+    data_type : str
+        Data type of the processed samples
+    investigation_type: str, optional
+        Existing investigation type to attach to the prep template
+    user_defined_investigation_type: str, optional
+        Existing user added investigation type to attach to the prep template
+    new_investigation_type: str, optional
+        Investigation type to add to the system
+
+    Returns
+    -------
+    dict of str
+        {'status': status,
+         'message': message,
+         'file': prep_template}
+    """
+    access_error = check_access(study_id, user_id)
+    if access_error:
+        return access_error
+    fp_rpt = check_fp(study_id, prep_template)
+    if fp_rpt['status'] != 'success':
+        # Unknown filepath, so return the error message
+        return fp_rpt
+    fp_rpt = fp_rpt['file']
+
+    # Add new investigation type if needed
+    investigation_type = _process_investigation_type(
+        investigation_type, user_defined_investigation_type,
+        new_investigation_type)
+
+    msg = ''
+    status = 'success'
+    try:
+        with warnings.catch_warnings(record=True) as warns:
+            # force all warnings to always be triggered
+            warnings.simplefilter("always")
```

Comment:
I don't think this needs to be here. Note that below you have a similar function and you are not calling `warnings.simplefilter`. The reason for that is because the system is already set up so by default the warnings are always triggered. Please remove as it is not needed.